madanalysis/configuration/weight_configuration.py

```diff
@@ -0,0 +1,331 @@
+################################################################################
+#
+#  Copyright (C) 2012-2023 Jack Araz, Eric Conte & Benjamin Fuks
+#  The MadAnalysis development team, email: <ma5team@iphc.cnrs.fr>
+#
+#  This file is part of MadAnalysis 5.
+#  Official website: <https://github.com/MadAnalysis/madanalysis5>
+#
+#  MadAnalysis 5 is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  MadAnalysis 5 is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with MadAnalysis 5. If not, see <http://www.gnu.org/licenses/>
+#
+################################################################################
+
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Text, Tuple
+
+import numpy as np
+
+
+@dataclass
+class Weight:
+    """Weight identifier class"""
```

Comment:
In the comments, we may need to write something that the code is meant to work with MG5_aMC weight handling (and conventions) only for now. We should test this on Pythia-only later on, and maybe other event generators... but one thing at a time!